planetary-social/nos
# Add feature flag and unit test for new media display

## Issues covered
#1176

## Description
Adds a feature flag for enabling the new media display which is not yet implemented. Establishes a pattern for using feature flags in the app. Adds a test to demonstrate how feature flags can be used and configured differently for testing and production.

## How to test
1. Create a post with media, or just find one
2. Ensure the post shows the media the same way it always did
3. Navigate to settings
4. Under Debug, enable the switch for "Enable new media display"
5. Find a post with media
6. Ensure it displays no gallery since there's no new media display yet

Additionally, run the unit tests to make sure the new one passes.

## Video
Here's how you can toggle the flag on and off. Only available in DEBUG builds.

https://github.com/planetary-social/nos/assets/59564/e32a4f80-bce8-4d2f-a149-941351b99186
